FEATURE: Make MultiSelectBox drag&drop sorting feature optional

laurahaenel opened the pull request:
<!--
Thanks for your contribution, we appreciate it!

ATTENTION: ALL NEW FEATURE PRs SHOULD TARGET THE MASTER BRANCH (bugfixes go to the least maintained branch)
-->

Solves #3715 .

**What I did**

- MultiSelectBox will default to non-dragging behavior.

**How I did it**

- Added new prop for the MultiSelectBox to ALLOW dragging.

**How to verify it**

- E.g. AssetEditor add prop `allowDragging={true}` (`Packages/Application/Neos.Neos.Ui/packages/neos-ui-editors/src/Editors/AssetEditor/index.js`)

With `allowDragging={false}`:
<img width="321" alt="image" src="https://github.com/user-attachments/assets/bfa9b01c-02ef-4f42-b1b1-e52939b71363">

Default:
<img width="316" alt="image" src="https://github.com/user-attachments/assets/f467fe30-17c6-45de-bacd-06bc4c13ed7e">


<!--
If possible, a screenshot or a gif comparing the new and old behavior would be great.
-->

patricekaufmann committed:
BUGFIX: fix specific cases where the requested croparea would be larger than the image itself

Commit:
Fix #3520 SelectBox invalid values

Commit:
ADD english translation

Christian-Pansch committed:
increase onChange debounce time and trigger onChange on FocusLos

ahaeslich committed:
Update packages/neos-ui-ckeditor5-bindings/src/ckEditorApi.js

Co-authored-by: Wilhelm Behncke <2522299+grebaldi@users.noreply.github.com>

Sebobo committed:
BUGFIX: Improve loading performance of UI and remove script tag with nodedata

With this change the minimal required nodedata for each node in the rendered
content is inserted as data attribute and not as inline script anymore.
This improves performance as no extra function call is executed for each node.

Additonally the reduction in rendered node attributes reduces the output filesize and again improves loading time.

To prevent just-in-time loading of nodes all incomplete nodedata is requested after the guest frame has finished loading.

Sebobo committed:
TASK: Cache user locale

Sebobo committed:
TASK: Don’t render nodedata to content and just load on initialisation

Sebobo committed:
TASK: Optimise node context resolution from flow query

mhsdesign committed:
TASK: Fix e2e

https://neos-project.slack.com/archives/C04PYL8H3/p1715701489862359?thread_ts=1715694569.690899&cid=C04PYL8H3

mhsdesign committed:
Followup #3776 set version field to empty string so it can be bumped

Commit:
TASK: Translated using Weblate (Dutch)

Currently translated at 100.0% (121 of 121 strings)

Translation: Neos/Neos.Ui - Main - 8.3
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/nl/

Commit:
TASK: Translated using Weblate (Spanish)

Currently translated at 100.0% (121 of 121 strings)

Translation: Neos/Neos.Ui - Main - 8.3
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/es/

bbehrendt-mm committed:
BUGFIX: findNodeToEdit() checks if the found node belongs to the correct site

Fixes #3795

Neos uses shared sessions when working with multiple sites on different domains. This fix ensures that we do not try to open a lastVisitedNode within a site the node does not belong to.

dr-format-d committed:
BUGFIX: `LinkEditor` placeholder now uses the custom placeholder correctly

Initially the user could set a placeholder, but inside the `LinkInput.js` it wouldn't be used and an unnecessary static placeholder would be used instead

Fixes #3797

alexandergirod committed:
TASK: Translated using Weblate (German)

Currently translated at 100.0% (121 of 121 strings)

Translation: Neos/Neos.Ui - Main - 8.3
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/de/

dr-format-d committed:
BUGFIX: `LinkEditor` placeholder now also displays default placeholder on empty `placeholder:` setting

If someone leaves the `placeholder:` setting empty, the default placeholder, instead of a completely empty `LinkEditor`-input will be shown.

Fixes #3797

Sebobo committed:
Merge pull request #3799 from dr-format-d/8.3

BUGFIX: `LinkEditor` placeholder now uses the custom placeholder correctly

mhsdesign committed:
Remove outdated comment

weblate committed:
TASK: Update translation files

Updated by "Remove blank strings" hook in Weblate.

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/
Translation: Neos/Neos.Ui - Main - 8.3

Sebobo committed:
Update packages/neos-ui-guest-frame/src/initializeGuestFrame.js

Co-authored-by: Marc Henry Schultz <85400359+mhsdesign@users.noreply.github.com>

Sebobo committed:
TASK: Load full nodes from content tree and skip them from loading for guest frame

The content tree is loaded early and can load fully loaded nodes without
being slower as the response time is the same. But this way we can skip
lots of nodes from being loaded by the guest frame if they are already
present in the store.

Sebobo committed:
TASK: Filter node duplicates in UI flow queries

bbehrendt-mm committed:
Update Classes/Controller/BackendController.php

Co-authored-by: Marc Henry Schultz <85400359+mhsdesign@users.noreply.github.com>

mhsdesign committed:
Merge pull request #3796 from bbehrendt-mm/8.3

BUGFIX: findNodeToEdit() checks if the node belongs to the correct site

mhsdesign committed:
Merge remote-tracking branch 'origin/8.3' into SelectBoxInvalidValues

mhsdesign committed:
WIP: Add processing of invalid values to `processSelectBoxOptions` to create schema for invalid entries

mhsdesign committed:
TASK: Simplify and improve `processSelectBoxOptions`

Sebobo committed:
BUGFIX: Let document tree searchbox scale with sidebar width

mhsdesign committed:
TASK: Rename SelectBoxHelpers.js and refactor to typescript

mhsdesign committed:
TASK: Add tests for `processSelectBoxOptions`

mhsdesign committed:
TASK: Add further tests for `processSelectBoxOptions` to test edgecases like filtering

Sebobo committed:
Merge pull request #3809 from neos/bugfix/searchbox-width

BUGFIX: Let document tree searchbox scale with sidebar width

mhsdesign committed:
BUGFIX: Handle empty string as empty value and dont generate selexbox option

mhsdesign committed:
BUGFIX: Fix sorting in multi selectbox if a value doesnt exist

The bool filter here is broken. we must keep the real index see daggableValues[dragIndex];:

https://github.com/neos/neos-ui/issues/3520#issuecomment-2185969334

mhsdesign committed:
TASK: Improve comment regarding `isNil` in selectbox

Sebobo committed:
TASK: Don’t merge existing nodes back into the store

Commit:
TASK: Update selectBox snapshot to fix CI

Sebobo committed:
Merge pull request #3745 from 73nici/SelectBoxInvalidValues

BUGFIX: Select box shows invalid value

grebaldi committed:
TASK: Make tree node drop targets discoverable in E2E tests

The drop targets used to have `visibility: hidden` set via CSS. Testcafé
considers them absent, which is semantically correct.

This commit uses a different CSS tactic do make the drop targets
invisible, but "physically" present.

grebaldi committed:
TASK: Make document tree filter toggle button discoverable in E2E tests

grebaldi committed:
TASK: Make node tree filter select box reset button discoverable in E2E tests

grebaldi committed:
TASK: Make document tree search reset button discoverable in E2E tests

Commit:
TASK: Translated using Weblate (Spanish)

Currently translated at 100.0% (122 of 122 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/es/
Translation: Neos/Neos.Ui - Main - 8.3

grebaldi committed:
TASK: Add mechanism that allows to alter Settings during E2E test

For this, the E2E test distribution now exposes two endpoints:

- /test/write-additional-settings
- /test/remove-additional-settings

An E2E test can thus via `fetch` add settings to the current setup and
remove the altered settings on cleanup.

This became necessary, because the tests for the node tree preset
feature require explicit settings to configure node tree presets. By
default, node tree presets are inactive.

grebaldi committed:
TASK: Add E2E tests for node tree presets

grebaldi committed:
TASK: Add failing E2E test for #3816

grebaldi committed:
TASK: Add failing E2E test for #2583

grebaldi committed:
TASK: Add failing E2E test for #2800

grebaldi committed:
BUGFIX: Allow-List nodes in document tree when filters are active

fixes: #3861, #2583

mhsdesign committed:
BUGFIX fix neos ui version label

Resolves: #3781

grebaldi committed:
PATCH: Anticipate `visibleContextPaths` being `undefined` when in content tree

grebaldi committed:
BUGFIX: Add `baseNodeType` to move change when multiple nodes are dropped

refs: #2800

grebaldi committed:
TASK: Deprecate all singular node operations

Every singular node operation can be expressed as a "multiple nodes"
operation.

Redundancy in this area has lead to bugs.

grebaldi committed:
TASK: Add failing test for #2800 Part II

grebaldi committed:
BUGFIX: Track `visible` state in tree also when context paths are updated

fixes: #2800

grebaldi committed:
TASK: Add failing E2E test that shows broken page reload with active node tree preset

grebaldi committed:
BUGFIX: Perform search on `RELOAD_STATE` when node type filter or search term is set

grebaldi committed:
TASK: Add failing E2E test that show superfluous toggle handle

grebaldi committed:
BUGFIX: Ensure that filtered nodes provide no info about children that do not match filter

mhsdesign committed:
TASK: Decouple `build-production` from `build-subpackages`

grebaldi committed:
TASK: Increase wait times for text changes in E2E tests

grebaldi committed:
Merge pull request #3751 from Alvadda/bugfix/increaseDebounceTimeOnCKEditorForUpdateInterval

BUGFIX: Change update interval on CKEditor

Sebobo committed:
Merge pull request #3825 from neos/bugfix/3781-fix-neos-ui-version-label

BUGFIX Fix Neos.Ui version information in build script

grebaldi committed:
TASK: Distribute prefixed ids across DropDown* components used by SelectBox

This allows us to target every part of a select box specifically in E2E
tests.

laurahaenel committed:
TASK: introduce allowDragging prop on MultiSelectBox

laurahaenel committed:
TASK: introduce allowDragging prop on MultiSelectBox

mhsdesign commented on packages/react-ui-components/src/MultiSelectBox/multiSelectBox.js:
```diff
@@ -31,6 +31,11 @@ class MultiSelectBox extends PureComponent {
+        /**
+         * Allows dragging.
+         */
+        allowDragging: PropTypes.bool,
```
im not totally found of the `allow*` naming schema, but it seems were following the pattern of `allowEmpty` so it makes sense :D

laurahaenel replied:
Could also be `enableDragging` if you like that more?

mhsdesign replied:
no thanks I was just reflecting

mhsdesign reviewed (commented):
Thanks for this additional option. This really makes sense for things that are unordered either way! :D

Though im not sure about the breakiness here ... the pr is not declared as breaking (no `!!!` in the title) ... and technically it is not allowed to be breaking for a minor like 8.4 ... at least not too breaky but that is discussable.

But as you said
> MultiSelectBox will default to non-dragging behavior.

It is very much a new behaviour that might be unexpected at first and our users will have to look how to revert it.

I really like that we introduce a "positive" flag for the `MultiSelectBox` rather than negating stuff over and over but maybe it will make sense to introduce an `allowDragging=true` default for certain - if not all - editors that use the `MultiSelectBox` implicitly like the `Neos.Neos/Inspector/Editors/SelectBoxEditor`

A default for each editor can be set via this config:

https://github.com/neos/neos-development-collection/blob/fbc6ea95f05315dfaf2f7b58c6a398955a40177c/Neos.Neos/Configuration/Settings.yaml#L283-L284

Wdyt?


Edit: @Sebobo wrote

> But this should be optional as not every use case needs this and then we have a feature that only confuses users.


so maybe this is one of these slighly breaking changes that is allowed in a minor :D

Commit:
TASK: Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (122 of 122 strings)

Co-authored-by: Neophyte <neophyte@users.noreply.hosted.weblate.org>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/pt_PT/
Translation: Neos/Neos.Neos.Ui - Main - 8.4

Commit:
TASK: Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (122 of 122 strings)

Co-authored-by: Neophyte <neophyte@users.noreply.hosted.weblate.org>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/pt_PT/
Translation: Neos/Neos.Ui - Main - 8.3

Sebobo commented:
Thx so much for the feature!

If you disable it by default this will break at least one of my plugins. Though in the Media UI I will finally be able to turn it off where it had no effect.

So I'm pro default "on".

mhsdesign committed:
TASK: Minimal cosmetically cleanup in `UserLocaleService`

mhsdesign committed:
TASK: Dont pass `nodes` explicitly to `initializeContentDomNode` but use the store

previously it was cumbersome to collect all the `nodes` that are really on the page. But it should make no difference to do the lookup on a bigger set.

mhsdesign committed:
TASK: Add comment that we deduplicate the passed nodes to the flowquery endpoint

that means `neosUiDefaultNodes` only needs one param.
Technically we use a `Set` most times when we invoke the `q` endpoint but seb convinced me that this filtering doesnt hurt and is probably more sane :D

mhsdesign reviewed (approved):
Thänks

mhsdesign committed:
Merge remote-tracking branch 'origin/8.3' into 8.4

laurahaenel commented:
Mh so I assume every native Neos.Editor which makes use of the MultiSelectBox should then be defaulting to `allowDragging={true}`, did I get this right?

weblate committed:
TASK: Translated using Weblate (Portuguese (Portugal))

Currently translated at 99.1% (122 of 123 strings)

TASK: Translated using Weblate (Portuguese)

Currently translated at 99.1% (122 of 123 strings)

TASK: Translated using Weblate (Spanish)

Currently translated at 99.1% (122 of 123 strings)

TASK: Translated using Weblate (German)

Currently translated at 99.1% (122 of 123 strings)

Co-authored-by: Anonymous <noreply@weblate.org>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/de/
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/es/
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/pt/
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/pt_PT/
Translation: Neos/Neos.Neos.Ui - Main - 8.4

juanlira committed:
TASK: Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (122 of 122 strings)

TASK: Translated using Weblate (Portuguese)

Currently translated at 100.0% (122 of 122 strings)

Co-authored-by: Neophyte <electraocareca@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/pt/
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/pt_PT/
Translation: Neos/Neos.Neos.Ui - Main - 8.4

juanlira committed:
TASK: Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (123 of 123 strings)

TASK: Translated using Weblate (Portuguese)

Currently translated at 100.0% (123 of 123 strings)

Co-authored-by: Neophyte <electraocareca@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/pt/
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/pt_PT/
Translation: Neos/Neos.Neos.Ui - Main - 8.4

juanlira committed:
TASK: Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (122 of 122 strings)

TASK: Translated using Weblate (Portuguese)

Currently translated at 100.0% (122 of 122 strings)

Co-authored-by: Neophyte <electraocareca@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/pt/
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/pt_PT/
Translation: Neos/Neos.Ui - Main - 8.3

Sebobo commented:
> Mh so I assume every native Neos.Editor which makes use of the MultiSelectBox should then be defaulting to `allowDragging={true}`, did I get this right?

The components default should be `true`. Then we don't need to set it on the editors. 
As this component is used in 3rd party plugins, we cannot change the default right now.

grebaldi committed:
BUGFIX: Prevent unneccessary server calls when inline editor is unchanged

Commit:
TASK: Translated using Weblate (Spanish)

Currently translated at 100.0% (123 of 123 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-neos-ui-main-84/es/
Translation: Neos/Neos.Neos.Ui - Main - 8.4

Commit:
TASK: Translated using Weblate (Spanish)

Currently translated at 100.0% (122 of 122 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/neos/neos-ui-main-8-3/es/
Translation: Neos/Neos.Ui - Main - 8.3

grebaldi committed:
Merge pull request #3833 from grebaldi/bugfix/3832/prevent-remote-calls-when-inline-editor-is-unchanged

BUGFIX: Prevent unneccessary server calls when inline editor is unchanged

Sebobo reviewed (changes requested):
Please add `allowDragging: true` to the defaultProps of the multiselectbox

laurahaenel committed:
TASK: set default value for allowDragging

laurahaenel committed:
TASK: set default value for allowDragging

Sebobo reviewed (changes requested):
Thx, the tests still need some adjustments though.

Sebobo committed:
Merge pull request #3822 from neos/bugfix/node-tree-presets-inconsistencies

BUGFIX: Fix inconsistencies in document tree with node tree presets

Sebobo committed:
TASK: Fix styleci issue

laurahaenel committed:
TASK: update snapshot

laurahaenel commented:
What about the circleci test? I guess I dont get why it fails

mhsdesign committed:
Merge pull request #3578 from patricekaufmann/bugfix/pixelsnapping-crop-overflow

BUGFIX: fix bug with pixelSnapping on certain aspectRatios and image dimensions

Sebobo commented:
It complains that it cannot see an editor field, as the select box covers it 

```
Warnings (1):
 --
  TestCafe cannot interact with the <label for="__neos__editor__property---uriPathSegment" class="Z9_R_G_label _49YZqG_reset hN5w1W_envelope__label">...</label> element because another element obstructs it.
  When something overlaps the action target, TestCafe performs the action with the topmost element at the original target's location.
  The following element with a greater z-order replaced the original action target: <div class="jYGwrW_listPreviewElement _49YZqG_reset jYGwrW_listPreviewElement--isHighlighted" role="button">...</div>.
  Review your code to prevent this behavior.
```

Could be also unrelated. Did you rebase recently?

Sebobo committed:
BUGFIX: Make sure to have the latest content when switching pages

This prevents showing outdated content, as the guest frame will not load new content by itself anymore.

mhsdesign committed:
Merge remote-tracking branch 'origin/8.3' into 8.4

mhsdesign committed:
BUGFIX: Always bundle custom svgs instead of using `resource://` paths

Sebobo committed:
BUGFIX: Hide preview central group if empty

mhsdesign committed:
Merge pull request #3770 from neos/bugfix/remove-ui-script-tag

BUGFIX: Improve UI loading performance by removing UI nodedata script tag and using less data

Sebobo committed:
Merge pull request #3837 from neos/bugfix/hide-empty-preview-central

BUGFIX: Hide preview central group if empty

mhsdesign committed:
BUGFIX: Include the image editors dummy image in the bundle

using `/_Resources/Static/Packages/Neos.Neos/Images/dummy-image.svg` is a bad bet as for cloud hosted resources this wont work and it will show a dead-fallback image.

mhsdesign committed:
TASK: Optimize Neos logo.svg

see discussion in Neos.Neos where we did the same https://github.com/neos/neos-development-collection/pull/5126/files#r1625874465

- remove obsolete `g` tags (and german illustrator references :D)
- inline style

mhsdesign committed:
Merge remote-tracking branch 'origin/8.3' into 8.4

mhsdesign committed:
Merge pull request #3836 from neos/bugfix/3616-bundle-custom-icons-insteadof-linking

BUGFIX: Always bundle custom svgs instead of using `resource://` paths

mhsdesign committed:
Revert "BUGFIX: Prevent unneccessary server calls when inline editor is unchanged"

This reverts commit 180906da9f6128fce322c25a17fd03f51e05b305.

The change introduced with pr https://github.com/neos/neos-ui/pull/3833 is not needed when onChange is only triggerd on real actual changes.

mhsdesign committed:
BUGFIX: 3839 & 3832 only issue changes if ckeditor is dirty

The dirty state can be tracked via the callback see https://github.com/ckeditor/ckeditor5/issues/996#issuecomment-407360730

mhsdesign committed:
BUGFIX: 3839 & 3832 simplify change persistence by using .flush();

... and thus we can avoid the isDirty state and second change:data callback

> a flush method to immediately invoke them

mhsdesign committed:
BUGFIX: Remove duplicate focus ckeditor border

With https://github.com/neos/neos-development-collection/pull/3618 the legacy ember css class `.neos-inline-editable` was removed (previously set here: https://github.com/neos/neos-development-collection/blob/881675b91185cbd8c41cb5e64f7467bf78710e04/Neos.Neos/Classes/Service/ContentElementEditableService.php#L69)

This leads to the ckeditor focus outline styling not being removable.

Instead, we use `[data-__neos-property][contenteditable]` to target ckeditor inline elements now.

Sebobo committed:
BUGFIX: Remove required flag from showResetButton prop on selectbox

The prop is actually not required and the UI tends to throw warnings because of the required flag.

mhsdesign committed:
Merge pull request #3848 from neos/bugfix/proptypes-warning-showresetbutton

BUGFIX: Remove required flag from showResetButton prop on selectbox

mhsdesign committed:
Merge pull request #3846 from neos/bugfix/3839-ckeditor-only-save-changes-if-dirty

BUGFIX: 3839 ckeditor only save changes if (really) dirty

mhsdesign committed:
Merge pull request #3847 from neos/bugfix/remove-ckeditor-focus-outline-css

BUGFIX: Remove duplicate focus ckeditor border

skurfuerst committed:
TASK: TestCafe Update

skurfuerst committed:
BUGFIX: ensure DB in Docker TestCafe Setup boots up again

before this change, we got a DB migration error (presumably because of too new MySQL version)

skurfuerst committed:
BUGFIX: fix TestCafe Testcases (part 1)

skurfuerst committed:
BUGFIX: change some errors / warnings during TestCafe run

skurfuerst committed:
BUGFIX: fix TestCafe Testcases (part 2)

skurfuerst committed:
BUGFIX: fix TestCafe Testcases (part 3)

skurfuerst committed:
BUGFIX: make setState handling more robust in ImageEditor to prevent warning if unmounted (needed to have the tests run through)

skurfuerst committed:
Merge pull request #3850 from neos/84-fix-e2e-tests

8.4: fix e2e tests

skurfuerst committed:
Merge pull request #3850 from neos/84-fix-e2e-tests

8.4: fix e2e tests
# Conflicts:
#	packages/neos-ui-redux-store/src/CR/Nodes/selectors.ts

markusguenther committed:
TASK: Use saucectl with configuration

markusguenther committed:
TASK: Adds missing API version

markusguenther committed:
TASK: Adjust test suite pathes

markusguenther committed:
TASK: Test test config per dimensions

Also configure a tunnel.

markusguenther committed:
TASK: Adds Sauce Connect Proxy 5 to CircleCI

markusguenther committed:
BUGIX: Adds missing region

markusguenther committed:
BUGFIX: Move tunnel config

The validation of the saucelabs config failed

markusguenther committed:
TASK: Add .sauceignore file and move tunnel start

markusguenther committed:
BUGFIX: Add missing npm dependencies

markusguenther committed:
TASK: Adjust .sauceignore and remove node module

Resolves: ERR failed to execute run command error="'node_modules' is ignored by sauceignore, but you have npm dependencies defined in your project; please remove 'node_modules' from your sauceignore file"

markusguenther committed:
BUGFIX: Allow localhost proxy

Resolves: sauce_connect proxying is denied to host "127.0.0.1:8081" localhost proxying is disabled

markusguenther committed:
TASK: Show console.logs

markusguenther committed:
TASK: Tests 1 dimensional tests on Mac and Windows 10

markusguenther committed:
TASK: Limit how many suites are executed at the same time.

markusguenther committed:
TASK: Add screenResolution as tests on mac could not resize

markusguenther committed:
TASK: Change MacOS screen size

As the resolution we specified  for Mac is not available.

markusguenther committed:
TASK: Enable smart retries when a test failed

mhsdesign committed:
BUGFIX: Neos logo icon width

Regression from https://github.com/neos/neos-ui/pull/3836

mhsdesign committed:
TASK: Followup to #3825

Currently, it says:
> UI version: origin/8.3-dev

but it should be:
> UI version: 8.3-dev

markusguenther committed:
TASK: Change reporting config and fix dimension path

markusguenther committed:
TASK: Remove feature to comment PR

Removes the post-acceptance-tests-recordings in CircleCI and the related script.

markusguenther committed:
TASK: Add more files to artifact download

Sebobo committed:
Merge pull request #3852 from neos/bugfix/neos-logo-icon-width

BUGFIX: Neos logo icon width

markusguenther committed:
TASK: Fetch target branch and use it as saucelabs tag

markusguenther committed:
TASK: Install newer version of GitHub CLI

markusguenther committed:
TASK: Passthru the target branch

skurfuerst committed:
Merge pull request #3851 from neos/task/use-saucectl

TASK: Use saucectl with configuration

skurfuerst committed:
Merge pull request #3851 from neos/task/use-saucectl

TASK: Use saucectl with configuration

markusguenther committed:
BUGFIX: Use CIRCLE_BRANCH when we are not triggered by a PR (#3854)

markusguenther committed:
BUGFIX: Use CIRCLE_BRANCH when we are not triggered by a PR (#3854)

markusguenther committed:
TASK: Refactor script to functions

To be able to reuse parts the change extracts tasks into functions.

markusguenther committed:
TASK: Unify shell scripts for local and saucelabs testing

Sebobo committed:
BUGFIX: Prevent augmenter from applying data of multiple nodes into the same element

This fixes a regression introduced in #b56135a01ecf59ae3a4990e3fd54ac766732e0e6 which
removed the script tag, causing the augmenter to add the data of multiple nodes into the same
html element in certain cases instead of adding an outer div.

With this change this behaviour is now more explicit instead of relying on the inner workings
of the augmenter.

kitsunet committed:
Merge pull request #3856 from neos/bugfix/enforce-dom-node-for-augmentation

BUGFIX: Prevent augmenter from applying data of multiple nodes into the same element

markusguenther committed:
BUGFIX: Trim leading and tailing whitespaces

Whitespaces like a tab can lead to a 404 error if a URL has been pasted to a LinkEditor or LinkInput.

As datasources, assets, URLs and slugs should not start with a whitespace, we can remove them.

Fixes: #3859

markusguenther committed:
BUGFIX: Prevent potential js error

Ensure that searchTerm is a string before trimming the value.

mhsdesign committed:
TASK: Run tests also on chrome in ci

mhsdesign committed:
BUGFIX: Stabilise test locally to have deterministic inspector groups positioning

When the inspector tab panel is scrolled just enough, so that there\'s enough space, SelectBox contents jump below the SelectBox.

Would fail otherwise because "document" would be before "tests"

mhsdesign committed:
WIP attempt to fix screen size for mac e2e tests

 ✖ Select Boxes - SelectBox opens above in creation dialog if there's not enough space below.

   1) Unable to resize the window because the specified size exceeds the screen size. On macOS, a window cannot be larger than the screen.

      Browser: Chrome 129.0.0.0 / Ventura 13

         21 |        .gt(await ReactSelector('NodeCreationDialog SelectBox').getBoundingClientRectProperty('top'));
         22 |});
         23 |
         24 |test('SelectBox opens above in creation dialog if there\'s not enough space below.', async t => {
         25 |    await t
       > 26 |        .resizeWindow(1200, 768)

mhsdesign committed:
TASK: Cleanup bash script

mhsdesign committed:
TASK: Add exit 1 handling in case of failed test suite

mhsdesign committed:
wip try a valid screen size

> WRN Suite errored. error="job start failed (404): Misconfigured -- Resolution you specified is not available for this OS/browser/version/device combo: OS: 'Mac 13', Browser: 'googlechrome', Version: '129.0.6668.59', Device: 'unspecified', Screen Resolution: '1920x1080'; available resolutions: '1024x768', '1152x864', '1280x960', '1376x1032', '1440x900', '1600x1200', '1920x1440', '2048x1536'"

mhsdesign committed:
TASK: remove retry for now in e2e

The retrys just fail:

Sauce TestCafe Runner 3.9.0
2Running TestCafe 3.6.2

8Error: Cannot find module '/Volumes/Sauce/node/20.14.0/node-v20.14.0-darwin-x64/bin/node_modules/npm/bin/npm-cli.js'

25
26Error: Cannot find module '/Volumes/Sauce/node/20.14.0/node-v20.14.0-darwin-x64/bin/node_modules/npm/bin/npm-cli.js'

56ERROR No tests match your filter.
57See https://testcafe.io/documentation/402638/reference/configuration-file#filter.
58
59Type "testcafe -h" for help.
60JUnit file generation skipped: no test suites detected.

mhsdesign committed:
wip reset screen size

mhsdesign committed:
wip revert test resize hack `resizeWindow` as mac os on e2e will fail

see 20dfb5a6c50b85e8975733b7b49480d753f9e416
https://github.com/neos/neos-ui/pull/3275

pKallert committed:
Fix: Sizing of alert buttons

mhsdesign committed:
give up

mhsdesign committed:
Merge pull request #3855 from neos/backport-e2e-tests

TASK: Backport the end to end tests from 8.4

mhsdesign committed:
Merge remote-tracking branch 'origin/8.3' into 8.4

mhsdesign committed:
Merge pull request #3861 from pKallert/fix-icons-flashmessage

FIX: Sizing of icons in alerts

JamesAlias committed:
TASK: Add docs for contribution getting started (#3798)

* TASK: Add docs for contribution getting started

* TASK: Adds code-style

---------

Co-authored-by: Markus Günther <info@unikka.de>

crydotsnake commented:
Ping @laurahaenel

laurahaenel commented:
Mh I cant rerun the circle ci tests (no permissions). I tried to run them locally but the setup is not correct on my machine. I will have a deeper look into it to get the same (failing) test results. Or should I just rebase?

markusguenther committed:
TASK: Assign trimmed value to the searchTerm

markusguenther committed:
TASK: just trim the leading part of a string

markusguenther committed:
TASK: Trim search term on manuall apply for

mhsdesign committed:
BUGFIX: Restore old Placeholder behavior to keep shown when focused (#3864)

* BUGFIX: Restore old Placeholder behavior to keep shown when focused

In Neos 7.3.17 and before the placeholder in ckeditor would not disappear when focused. And only when typing. By switching to the native placeholder feature of ckeditor in https://github.com/neos/neos-ui/pull/3558 we introduced a regression as that implementation would hide the placeholder while just clicking into.
This is especially noticeable for centered texts as the cursor needs one millisecond to adjust and its glitchy.

 This will eventually be fixed in Ckeditor 24 something. As a hotfix for our version 16 i introduce the patch of https://github.com/ckeditor/ckeditor5/pull/8474 (without the css absolute change as that would make centered text uneditable). The mentioned fix was not merged directly into ckeditor but made way more complicated: https://github.com/ckeditor/ckeditor5/pull/8867
 We dont need the more complicated fix as we dont use the mentioned features of the placeholder in a image description and such.

* TASK: Move cursor before the placeholer

* TASK: Hide break in placeholder context

To move the cursor to the first position, the placeholder text is moved backwards. Therefore, a change with a break is disadvantageous.

---------

Co-authored-by: Markus Günther <info@unikka.de>

laurahaenel committed:
Merge remote-tracking branch 'refs/remotes/fork/feature/3715-Make-MultiSelectBox-drag&drop-sorting-feature-optional' into feature/3715-Make-MultiSelectBox-drag&drop-sorting-feature-optional

laurahaenel commented:
I think I will make a new branch, this one is outdated and neither rebase nor merge is feasible 😅 Cherry-picking it is
<img width="157" alt="image" src="https://github.com/user-attachments/assets/3838ea4e-c939-46d6-8063-8a03761fa14b">
I also broke the history when I forced it 🙄 Not using the outdated branch.

This is the new MR https://github.com/neos/neos-ui/pull/3874
